Sources/OktaDirectAuth/Internal/Authentication Factors/PrimaryFactor.swift

```diff
@@ -26,61 +26,73 @@ extension DirectAuthenticationFlow.PrimaryFactor {
         switch self {
         case .otp: fallthrough
         case .password:
             let request = TokenRequest(openIdConfiguration: openIdConfiguration,
                                        clientConfiguration: flow.client.configuration,
+                                       currentStatus: currentStatus,
                                        loginHint: loginHint,
                                        factor: factor,
                                        grantTypesSupported: flow.supportedGrantTypes)
             return TokenStepHandler(flow: flow, request: request)
         case .oob(channel: let channel):
+            var bindingContext: DirectAuthenticationFlow.BindingUpdateContext?
+            if case .bindingUpdate(let context) = currentStatus {
+                bindingContext = context
+            }
             return try OOBStepHandler(flow: flow,
                                       openIdConfiguration: openIdConfiguration,
+                                      currentStatus: currentStatus,
                                       loginHint: loginHint,
-                                      mfaToken: currentStatus?.mfaToken,
                                       channel: channel,
                                       factor: factor,
                                       bindingContext: bindingContext)
+        case .webAuthn:
+            let mfaContext = currentStatus?.mfaContext
+            let request = try WebAuthnChallengeRequest(openIdConfiguration: openIdConfiguration,
+                                                       clientConfiguration: flow.client.configuration,
+                                                       loginHint: loginHint,
+                                                       mfaToken: mfaContext?.mfaToken)
+            return ChallengeStepHandler(flow: flow, request: request) {
+                .webAuthn(.init(request: $0,
+                                mfaContext: mfaContext))
+            }
         }
     }
     
-    var tokenParameters: [String: Any]? {
+    func tokenParameters(currentStatus: DirectAuthenticationFlow.Status?) -> [String: String] {
+        var result: [String: String] = [
+            "grant_type": grantType(currentStatus: currentStatus).rawValue,
+        ]
+        
         switch self {
         case .otp(code: let code):
-            return [
-                "grant_type": grantType.rawValue,
-                "otp": code
-            ]
+            result["otp"] = code
         case .password(let password):
-            return [
-                "grant_type": grantType.rawValue,
-                "password": password
-            ]
-        case .oob:
-            return nil
+            result["password"] = password
+        case .oob: break
+        case .webAuthn: break
         }
-
+        
+        return result
     }
 
-    var grantType: GrantType {
+    func grantType(currentStatus: DirectAuthenticationFlow.Status?) -> GrantType {
```

Comment:
Does this need currentStatus as a function parameter? I don't see that being used in this function

Reply:
The `grantType` property/function is a property of the `AuthenticationFactor` protocol, which is shared by both Primary and Secondary factor types.  The `SecondaryFactor` uses the current status to determine whether or not the grant type should be a primary authentication, or MFA.  

For example, [in this line](https://github.com/okta/okta-mobile-swift/pull/172/files#diff-68b991a9abe502d7dc9f8d2179080fb83e6bd0971e3d84e5ed10968d7fc80a02R93) the code checks to see if the current status is associated with an MFA Token, which changes which grant type should be used.